### PR TITLE
Create a document with the database schema

### PR DESCRIPTION
**Pull request by MichaelKim20**
I created a document with the database schema
This includes schemas and scripts for the database tables required by the API server.

### CODE REVIEW

**Comment by TrustHenry** [comment]
Nice Work!
Additional tables are required.
A table is required to summary a list of validators for each block height.

**Comment by MichaelKim20** [comment]
> Nice Work!
> Additional tables are required.
> A table is required to summary a list of validators for each block height.

Yes, that's a very good point. I'll add it.

File: docs/Database_Schema.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+| random_seed       | TEXT      |    | Y        |          ||
+| cycle_length      | INTEGER   |    | Y        |          ||
+| enroll_sig        | TEXT      |    | Y        |          ||
+
+### _Create Script_
+
+```sql
+CREATE TABLE "enrollments" (
+    "block_height"    INTEGER NOT NULL,
+    "enrollment_index"    INTEGER NOT NULL,
```
Comment (Geod24) [inline docs/Database_Schema.md:52]: What is `enrollment_index` ?

**Comment by TrustHenry** [comment]
In addition, it is recommended that you have an "Account balance table" by PublicKey in the recent block.

**Comment by MichaelKim20** [comment]
> In addition, it is recommended that you have an "Account balance table" by PublicKey in the recent block.

@TrustHenry : Good Point! I added it.

File: docs/Database_Schema.md (hunk 1 above)
Comment (Geod24) [reply docs/Database_Schema.md:52]: The index of this `Enrollment` in the block's `enrollments` array

File: docs/Database_Schema.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+## 7. Table **validators**
+
+### _Schema_
+
+| Column            | Data Type | PK | Not NULL | Default  |Description|
+|:----------------- |:--------- |:--:|:--------:| -------- | --------- |
+|  block_height     | INTEGER   | Y  | Y        |          | The height of the block|
+|  utxo_key         | TEXT      | Y  | Y        |          ||
+|  enrolled_at      | INTEGER   |    | Y        |          ||
```
Comment (Geod24) [inline docs/Database_Schema.md]:
```suggestion
|  enrolled_at     | INTEGER   | Y  | Y        |          | The height this validator enrolled at |
|  utxo_key         | TEXT      | Y  | Y        |          ||
```

File: docs/Database_Schema.md [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+### _Create Script_
+
+```sql
+CREATE TABLE "preimages" (
+	"utxo_key"	TEXT NOT NULL,
+	"preimage_distance"	INTEGER NOT NULL,
+	"preimage_hash"	INTEGER NOT NULL,
+	PRIMARY KEY("utxo_key")
+)
+```
```
Comment (Geod24) [inline docs/Database_Schema.md]: Remove

**Review by Geod24 — COMMENTED** [review]
Discussed IRL, nice work!

File: docs/Database_Schema.md [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+CREATE TABLE "accounts" (
+	"block_height"	INTEGER NOT NULL,
+	"address"	TEXT(56) NOT NULL,
+	"amount"	NUMERIC NOT NULL,
+	"spendable_amount"	NUMERIC NOT NULL,
+	"unspendable_amount"	NUMERIC NOT NULL
+	"is_validator"	TEXT(1) NOT NULL
+	PRIMARY KEY("block_height","address")
+)
+```
```
Comment (Geod24) [inline docs/Database_Schema.md]: Let's leave this out of the current sprint.

**Review by Geod24 — APPROVED** [review]

**Pull request merged**